Aptoide/aptoide-client-v8
# [APP-30] Out of space storage

**What does this PR do?**

This PR aims at implementing the new Out of Space dialog, on all views where downloads are performed and were showing the out of space error as a dialog. As it is written on the ticket, if there is not enough space to download the app, we try to clear the cache and resume the download. If clearing the cache is not enough, then we will show a list of apps and allow the user to uninstall them


**Database changed?**

   Yes

**Where should the reviewer start?**

- [ ] OutOfSpaceDialogFragment.kt

**How should this be manually tested?**

  In a device with low storage (you could reduce the memory of an emulator for this), try to perform a case where cache is the thing impeding the the instalation and another case where clearing the cache is not enough to install the app(bigger app).
Please make sure that besides the main feature, the analytics events and database are ok.

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-30](https://aptoide.atlassian.net/browse/APP-30)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 



**Code Review Checklist**
